<launch>
  <arg name="queue_size" default="30" />

  <group ns="stereo" >
    <group ns="left_camera" >
      <node name="apriltag_ros_continuous_node"
            pkg="apriltag_ros" type="apriltag_ros_continuous_node"
            clear_params="true"
            output="screen" >
        <remap from="image_rect" to="image_rect_color" />
        <remap from="camera_info" to="camera_info" />

        <param name="publish_tag_detections_image" type="bool" value="true" />
        <param name="queue_size" type="int" value="$(arg queue_size)" />
        <rosparam command="load" file="$(find apriltag_ros)/config/settings.yaml" />
        <rosparam command="load" file="$(find apriltag_ros)/config/tags.yaml" />
      </node>
    </group>

    <group ns="right_camera" >
      <node name="apriltag_ros_continuous_node"
            pkg="apriltag_ros" type="apriltag_ros_continuous_node"
            clear_params="true"
            output="screen" >
        <remap from="image_rect" to="image_rect_color" />
        <remap from="camera_info" to="camera_info" />
        <param name="publish_tag_detections_image" type="bool" value="true" />
        <param name="queue_size" type="int" value="$(arg queue_size)" />
        <rosparam command="load" file="$(find apriltag_ros)/config/settings.yaml" />
        <rosparam command="load" file="$(find apriltag_ros)/config/tags.yaml" />
      </node>
    </group>
  </group>

</launch>
